Backpack/FloatingNotification/Classes/FloatingNotificationViewModel.swift

```diff
@@ -0,0 +1,29 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+struct FloatingNotificationViewModel {
+    let parentView: UIView
+    let text: String
+    let buttonTitle: String?
+    let onTap: (() -> Void)?
```

Comment:
I'd suggest grouping data that needs to go together in a separate type (could be a nested struct here). this way the compiler does not allow inconsistent state to be represented, like creating a button with title but no action, or vice versa.

I'd create a FloatingNotificationViewModel.Action struct, containing a NON optional title and action, and then have the FloatingNotificationViewModel have an optional action.

Reply:
Yeah sure can do